inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -163,6 +188,119 @@ def update_total_penalty(self, task_id, username):
                                                       {"$set": {"total_penalty": new_penalty}
                                                        })
 
+    def on_change_task_submission_mode(self, courseid, task_id, submission_mode_change, task_hints):
+
+        "Update the user hints document structure when the task mode is changed."
+        "When a individually task, each student have it's own user hints document."
+        "When a group task, there is only a document for the students group"
+
+        if(submission_mode_change):  
+            
+            classrooms = self._database.aggregations.find({"courseid": courseid}
+            )
+
+            for classroom in classrooms:
+
+                for group in classroom.get("groups"):
+
+                    students = group.get("students")
+                    
+                    group_hints_document = {}
+                    if len(students) == 1:
+                        group_hints_document = self.get_user_hints(task_id, students[0])
+                    else:
+                        group_hints_document = self.get_user_hints(task_id, students)
```

Comment:
Why is this done? cannot you just send the students object always as the query is done like that? What if the number of students changes later? this will be wrong